### PR TITLE
Merge release/18.5 into develop

### PR DESCRIPTION
**Pull request by AliSoftware**
## What

 - [x] Final Release Notes
 - [X] Gutenberg 1.64.1 bugfix†
 - [X] Update to release-toolkit 2.0

## Conflict Resolution

Kept `develop` in the below `build.gradle` expected conflict, so that `develop` would still be on 1.65.0 and not 1.64.1 after this PR is merged:
```
<<<<<<< merge/release-18.5-into-develop
    ext.gutenbergMobileVersion = 'v1.64.1'
=======
    ext.gutenbergMobileVersion = 'v1.65.0-alpha1'
>>>>>>> develop
```

(†) which is why the actual fix that this beta was all about doesn't appear in this PR's diff.

### CODE REVIEW

**Review by oguzkocer — APPROVED** [review]
> Kept develop in the below build.gradle expected conflict, so that develop would still be on 1.65.0 and not 1.64.1 after this PR is merged

Actually in these cases we should have a new alpha tag (in this case `v1.65.0-alpha2`) from the current gutenberg-mobile develop so their hotfix would be included.

In practice, this probably doesn't make too big of a difference _in this case_, WPAndroid `develop` will just get the gutenberg-mobile hotfix a bit later. I am going to approve this PR, but @wordpress-mobile/owl-team you might want to keep this in mind going forward just so you don't get surprised by a build failure or a crash when you resolve the gb-mobile version to current develop. (after a hotfix)

**Comment by AliSoftware** [comment]
> Actually in these cases we should have a new alpha tag (in this case v1.65.0-alpha2) from the current gutenberg-mobile develop so their hotfix would be included.

Yep, that's usually what I expect to happen indeed (cc @geriux or @antonis will likely confirm)

Usually the Gutenberg team do that soon(-ish) _after_ they the fix in release: i.e. they do the `1.64.1` bugfix for the release branch first ([this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15484)) as it's the most urgent one to fix; then I do a new beta and merge `release` into `develop` (this PR)… and _then_, a bit later, they cherry-pick their fix to apply it in the current `1.65.0-alpha*` and merge a new `alpha*` in `develop` once ready. So that fix in `develop` usually happens _after_ they PRs like this one.

TL;DR: I do expect the Gutenberg team to indeed to a `v1.65.0-alpha2` release in `develop` in the upcoming days. At least that's usually the order of how things have happened in the past for these situation, and have worked for us so far 🙂

**Comment by oguzkocer** [comment]
@AliSoftware That all sounds good to me and in practice it should work most of the time. However, consider a case where there is a breaking change in the `react-native-bridge` library during a hotfix, such as adding a new parameter to an existing method, then this PR for merging the release back to develop will fail to build when we pick the gutenberg-mobile version from `develop`. Ideally, we'd have the new alpha tag ready at the same time as the hotfix, so that these PRs will use that new alpha tag. Since the combination of alpha tags & the hotfix is not that common, I think we can live with the non-ideal solution as long as we are all aware of this.

P.S: I do expect everyone doing releases to already know these and I am sorry for repeating it, but I think it's best to have it written somewhere just in case and it might end up helping someone understand the processes a bit better especially considering this is all open source.

**Comment by AliSoftware** [comment]
Ah that is a very good point!

Thanks for flagging and for the reminder 👍

**Comment by geriux** [comment]
> TL;DR: I do expect the Gutenberg team to indeed to a `v1.65.0-alpha2` release in `develop` in the upcoming days. At least that's usually the order of how things have happened in the past for these situation, and have worked for us so far 🙂

Hey there 👋 I've just created a [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15493) to update it.
